features/networking/multus.feature

```diff
@@ -1394,4 +1394,58 @@ Feature: Multus-CNI related scenarios
+      
+  # @author weliang@redhat.com
+  # @case_id OCP-31999
+  @admin
+  Scenario: Whereabouts with exclude IP address	
+    # Make sure that the multus is enabled
+    Given the multus is enabled on the cluster
+    # Create the net-attach-def via cluster admin
+    Given I have a project
+    Given I obtain test data file "networking/multus-cni/NetworkAttachmentDefinitions/whereabouts-excludeIP.yaml"
+    When I run oc create as admin over "whereabouts-excludeIP.yaml" replacing paths:
+      | ["metadata"]["namespace"] | <%= project.name %>          |
+    Then the step should succeed
+    
+    # Create a pod absorbing above net-attach-def
+    Given I obtain test data file "networking/multus-cni/Pods/generic_multus_pod.yaml"
+    When I run oc create over "generic_multus_pod.yaml" replacing paths:
+      | ["metadata"]["name"] | macvlan-bridge-whereabouts-pod1                              |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"] | whereabouts-excludeip  |
+      | ["spec"]["containers"][0]["name"] | whereabouts-excludeip                           |
+    Then the step should succeed
+    And the pod named "macvlan-bridge-whereabouts-pod1" becomes ready
+    # Check the created pod has correct ip 
+    When I execute on the "macvlan-bridge-whereabouts-pod1" pod:
+      | /usr/sbin/ip | a  |
+    Then the output should contain:
+      | 192.168.10.4      |
+
+    # Create second pod absorbing above net-attach-def
+    Given I obtain test data file "networking/multus-cni/Pods/generic_multus_pod.yaml"
+    When I run oc create over "generic_multus_pod.yaml" replacing paths:
+      | ["metadata"]["name"] | macvlan-bridge-whereabouts-pod2                              |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"] | whereabouts-excludeip  |
+      | ["spec"]["containers"][0]["name"] | whereabouts-excludeip                           |
+    Then the step should succeed
+    And the pod named "macvlan-bridge-whereabouts-pod2" becomes ready
+    # Check the created pod has correct ip 
+    When I execute on the "macvlan-bridge-whereabouts-pod2" pod:
+      | /usr/sbin/ip | a  |
+    Then the output should contain:
+      | 192.168.10.5      |
+      
+    # Create third pod absorbing above net-attach-def
+    Given I obtain test data file "networking/multus-cni/Pods/generic_multus_pod.yaml"
+    When I run oc create over "generic_multus_pod.yaml" replacing paths:
+      | ["metadata"]["name"] | macvlan-bridge-whereabouts-pod3                              |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"] | whereabouts-excludeip  |
+      | ["spec"]["containers"][0]["name"] | whereabouts-excludeip                           |
+    Then the step should succeed
+    And the pod named "macvlan-bridge-whereabouts-pod2" becomes ready
+    # Check the created pod should not has a ip
+    When I execute on the "macvlan-bridge-whereabouts-pod3" pod:
```

Comment:
@weliang1 This is not executing correctly. You might wanna investigate this

```[19:50:13] INFO> Shell Commands: oc exec macvlan-bridge-whereabouts-pod3  --kubeconfig=/home/weliang/workdir/weliang-weliang/ocp4_testuser-0.kubeconfig -n b35d5  -- /usr/sbin/ip a
      
      STDERR:
      error: unable to upgrade connection: container not found ("whereabouts-excludeip")
      
      [19:50:13] INFO> Exit Status: 1
      | /usr/sbin/ip | a |```

Reply:
@anuragthehatter good catuch,using wrong pod name, update to the step like:
 And the pod named "macvlan-bridge-whereabouts-pod3" status becomes :pending within 60 seconds
New testing log:
https://privatebin-it-iso.int.open.paas.redhat.com/?b5803b5b23c17536#h3At3mDCrvlgSZP0O28eheEUOoQIF/izLHyJfDoN27g=